opts/opts.go

```diff
@@ -486,3 +490,59 @@ func (m *MemSwapBytes) UnmarshalJSON(s []byte) error {
 	b := MemBytes(*m)
 	return b.UnmarshalJSON(s)
 }
+
+// GenericResource is a concept that a user can use to advertise user-defined
+// resources on a node and thus better place services based on these resources.
+// E.g: NVIDIA GPUs, Intel FPGAs, ...
+// See https://github.com/docker/swarmkit/blob/master/design/generic_resources.md
+type GenericResource []swarm.GenericResource
```

Comment:
This code (everything below) should go into `cli/command/service/opts.go` or even better `cli/command/service/generic_resource_opts.go`. The `opts` package is only for generic thigns that will be used across many command packages.